### ISSUE TITLE
[BUG] browserType.launch: Timeout 30000ms exceeded

### ISSUE THREAD

**Issue by bpasero** [microsoft/playwright#15127]
Extracted from https://github.com/microsoft/vscode/issues/153214

Is there a way for us to diagnose these issues without enabling verbose logs in playwright? I feel like playwright should enable by default the minimal amount of logs to diagnose issues like these. 

Here it seems as is a browser could not be launched on Windows, but interestingly previous tests that also require the same browser to launch succeed:

```
browserType.launch: Timeout 30000ms exceeded.
=========================== logs ===========================
<launching> C:\Users\cloudtest\AppData\Local\ms-playwright\firefox-1322\firefox\firefox.exe -no-remote -headless -profile C:\Users\CLOUDT~1\AppData\Local\Temp\playwright_firefoxdev_profile-byFYsg -juggler-pipe -silent
<launched> pid=8356
[pid=8356][err] *** You are running in headless mode.
[pid=8356][out] 
[pid=8356][out] Juggler listening to the pipe
============================================================
    at runTestsInBrowser (D:\a\_work\1\s\test\integration\browser\src\index.ts:36:48)
    at D:\a\_work\1\s\test\integration\browser\src\index.ts:184:9 {
  name: 'TimeoutError'
}
Exec: 
At D:\a\_work\1\s\build\azure-pipelines\win32\exec.ps1:22 char:3
##[error]PowerShell exited with code '1'.
+         throw ("Exec: " + $errorMessage)
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Exec: :String) [], RuntimeException
    + FullyQualifiedErrorId : Exec:
```

**Comment by pavelfeldman** [comment]
@[bpasero](https://github.com/bpasero) can you start with passing `3 * 60 * 1000` as a [browserType.launch({timeout})](https://playwright.dev/docs/next/api/class-browsertype#browser-type-launch-option-timeout)  to see if the action itself is under stress and can't run the browser within the given default timeout?

**Comment by aslushnikov** [comment]
> Is there a way for us to diagnose these issues without enabling verbose logs in playwright? 

It was probably a slow firefox startup, so it's not like logging would help it. Many things have landed in Firefox to address this and some other issues. These are available in Playwright canary version:

```bash
npm i @playwright/test@next
npx playwright install firefox
```

And these will be also released in v1.24 in 1-2 weeks.

I'll merge the whole issue in #14559
